### PR TITLE
Exporter refactor

### PR DESCRIPTION
**Pull request by chelm**
This is the first pass a bit of a cleaned up exporter. Its still a bit of mess i think and the logic needs to be rewritten, but at least not we have the minimum amount of code being duplicated across methods.

Also added the ability to support projections for shapefiles only, if a wkid is passed in.

### CODE REVIEW

**Comment by ngoldman** [comment]
Definitely big enough that it's hard to follow what's being refactored. Are there any breaking changes?

**Comment by ngoldman** [comment]
or just things that were exposed on the koop object that are no longer there or anything like that

**Comment by chelm** [comment]
@ngoldman yeah its huge, no breaking changes just a reworking of shared code. This is a work in progress for sure, but need to get the projection code out for our release next week.

**Comment by ngoldman** [comment]
:+1: ok just trying to keep up so I can contribute once I'm out of this never-ending multi-day multi-state meeting

**Comment by chelm** [comment]
grrr failing test!

**Comment by jwieringa** [comment]
If the failing test is related to a gdal dependency, take a look at how @scooterw is configuring Travis to install GDAL https://github.com/scooterw/ffi-ogr/blob/master/before_install.sh

Something like this in the travis config might do the trick?

```
before_install:
  - sudo apt-key adv --recv-keys --keyserver keyserver.ubuntu.com 16126D3A3E5C1192
  - sudo apt-get install python-software-properties -y
  - sudo add-apt-repository ppa:ubuntugis/ppa -y
  - sudo apt-get update -qq
  - sudo apt-get install libgdal-dev
```

**Comment by ngoldman** [comment]
@jwieringa++

**Comment by ngoldman** [comment]
ogr2ogr is probably something we can eventually bust out of core and test independently too

**Comment by chelm** [comment]
Thanks @jwieringa 
and yeah @ngoldman totally the next move. Just like tiles, exporting become a plugin. The move today was very much towards that but there are still some dependencies i need to re-think before the trigger it pulled. 

im gonna merge this...
